realm/src/main/java/io/realm/Realm.java

```diff
@@ -813,17 +809,21 @@ private static void initializeRealm(Realm realm) {
         }
         checkHasPrimaryKey(clazz);
 
-        JsonReader reader = new JsonReader(new InputStreamReader(in, "UTF-8"));
+        // As we need the primary key value we have to first parse the entire input stream as in the general
```

Comment:
It is not completely useless. It does make client code easier (ie. they don't have to manually have to write a Scanner and convert to a JSONObject). But I think general consensus is to keep it for now until we decide in what direction to move our JSON support?